src/contracts/reader/SettlementStorageReader.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+/// @title Gnosis Protocol v2 Settlement Storage Reader
+/// @author Gnosis Developers
+contract SettlementStorageReader {
+    // Storage Reader must have the same storage layout as the contract it reads.
+    // This sneaky storage member is inherited through ReentrancyGuard
+    uint256 private _status;
+
+    mapping(bytes => uint256) public filledAmount;
```

Comment:
Would it make sense to make this contract inherit the one that is read? The only downside I see is the extra deployment cost.

Reply:
Not keen on this idea as the settlement contract is already almost 3.5M gas. Is a nice idea, but I'd like the storage reader to stay as small as possible.

Reply:
I slightly prefer it this was as the reader contracts are intended to be "light". For a real reader contract, we probably want some code to ensure storage layouts match and not have to rely on inheritance.

Also worth noting that getting this wrong isn't super bad, it just means you lose the gas costs of the deployment and need to re-deploy with a fix (it can't clobber storage as its called with a `STATICCALL` and always causes a `REVERT`).